src/shared/install.c

```diff
@@ -52,18 +52,21 @@ typedef struct {
         OrderedHashmap *have_processed;
 } InstallContext;
 
-typedef enum {
-        PRESET_UNKNOWN,
-        PRESET_ENABLE,
-        PRESET_DISABLE,
-} PresetAction;
-
 struct UnitFilePresetRule {
         char *pattern;
         PresetAction action;
         char **instances;
 };
 
+static const char *const preset_action_table[_PRESET_ACTION_MAX] = {
+        [PRESET_UNKNOWN] = "unknown",
+        [PRESET_ENABLE]  = "enabled",
+        [PRESET_DISABLE] = "disabled",
+        [PRESET_IGNORE]  = "ignored",
```

Comment:
So there's a slight asymmetry here: the enum uses the infinitive, while the string uses the past tense. Which makes some sense, but maybe we should make this clearer. i.e. call the table preset_action_past_tense_table[] and the function preset_action_past_tense_to_string() hence?

and maybe add a comment here as extra bonus, "NB! strings use past tense"